### PR TITLE
PSR2 for CMS + linting

### PR DESCRIPTION
**Pull request by tractorcow**
Replaces https://github.com/silverstripe/silverstripe-cms/pull/1688

Instead of a tests/phpcs directory, I've put the coding standards into the root folder.

### CODE REVIEW

**Comment by tractorcow** [comment]
Rebased.

**Comment by chillu** [comment]
> Instead of a tests/phpcs directory, I've put the coding standards into the root folder.

Do we need to change this for the existing framework definitions as well to maintain consistency?

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ matrix:
 
 before_script:
  - composer self-update || true
+  - "if [ \"$PHPCS_TEST\" = \"1\" ]; then pyrus install pear/PHP_CodeSniffer; fi"
```
Comment (chillu) [inline .travis.yml]: Wrong indentation breaking the build, fixing now

Comment (tractorcow) [inline .travis.yml]: Doh, rebase error sorry.

**Comment by tractorcow** [comment]
> Do we need to change this for the existing framework definitions as well to maintain consistency?

Yes we should, but not urgent. :)

**Pull request merged**